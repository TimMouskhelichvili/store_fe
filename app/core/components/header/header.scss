@import 'core/styles/variables';

.header {
    width: 100%;
    height: 60px;
    background: $primary;
}