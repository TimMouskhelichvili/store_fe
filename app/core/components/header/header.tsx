import * as React from 'react';
import * as styles from 'app/core/components/header/header.scss';

import { Query, QueryResult } from 'react-apollo';

import gql from 'graphql-tag';

class Header extends React.Component {
    public render(): React.ReactNode {
        const header = (context: QueryResult): React.ReactNode => {
            if (context.loading) return <header className='noselect'></header>;

            return (
                <header className={styles.header}>
                    asdasdasdsd
                    <button onClick={(): void => {
                        context.client.writeData({
                            data: { sidemenu: true }
                        });
                    }}>aaaa</button>
                </header>
            );
        };

        return <Query query={gql`{ sidemenu @client }`}>{header}</Query>;
    }
}

export {
    Header
};
