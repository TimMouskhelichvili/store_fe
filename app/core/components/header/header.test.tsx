import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Header } from 'core/components/header/header';

describe('Core component header tests', (): void => {
    it('renders', (): void => {
        const component = renderer.create(<Header />);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
});
