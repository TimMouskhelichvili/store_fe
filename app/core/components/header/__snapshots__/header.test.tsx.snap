// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Core component header tests renders 1`] = `
<header
  className="header"
>
  asdasdasdsd
</header>
`;
